cvmfs/s3fanout.cc

```diff
@@ -897,17 +828,8 @@ Failures S3FanoutManager::InitializeRequest(JobInfo *info, CURL *handle) const {
     retval = curl_easy_setopt(handle, CURLOPT_NOBODY, 0);
     assert(retval == CURLE_OK);
     retval = curl_easy_setopt(handle, CURLOPT_INFILESIZE_LARGE,
-                              static_cast<curl_off_t>(info->payload_size));
+                              static_cast<curl_off_t>(info->origin->GetSize()));
```

Comment:
I kept it there specifically because of this line https://github.com/cvmfs/cvmfs/blob/devel/cvmfs/upload_s3.cc#L255 , where the origin does not exist anymore.

Reply:
Oh, fair enough.  Let's add this as a comment when `payload_size` is assigned.